fe1-web/src/features/witness/network/index.ts

```diff
@@ -1,18 +1,28 @@
-import { ActionType, MessageRegistry, ObjectType } from 'core/network/jsonrpc/messages';
+import { ActionType, ObjectType } from 'core/network/jsonrpc/messages';
+import { getStore } from 'core/redux';
 
+import { WitnessConfiguration } from '../interface';
 import { WitnessMessage } from './messages';
 import { handleWitnessMessage } from './WitnessHandler';
+import { afterMessageProcessingHandler, makeWitnessStoreWatcher } from './WitnessStoreWatcher';
 
 /**
  * Configures the network callbacks in a MessageRegistry.
  *
- * @param registry - The MessageRegistry where we want to add the mappings
+ * @param config - The witness feature configuration object
  */
-export function configureNetwork(registry: MessageRegistry) {
-  registry.add(
+export const configureNetwork = (config: WitnessConfiguration) => {
+  config.messageRegistry.add(
     ObjectType.MESSAGE,
     ActionType.WITNESS,
-    handleWitnessMessage,
+    handleWitnessMessage(config.getCurrentLao),
     WitnessMessage.fromJson,
   );
-}
+
+  // listen for new processable messages
+  const store = getStore();
+  store.subscribe(makeWitnessStoreWatcher(store, afterMessageProcessingHandler(config.enabled)));
+  /* config.messageRegistry.addAfterProcessingHandler(
+    afterMessageProcessingHandler(config.enabled, config.isLaoWitness),
+  ); */
```

Comment:
Did you leave this on purpose ?
Also, I'm not sure I understand the difference between those two, is it that if the handler fails, the entire message processing fails, when the subscribing way only "watches" (does not influence the processing logic even if it fails)?

Reply:
Oh whoops no, thank you!

Both approaches would have not make the message processing fail, the watcher approach is just a bit more "clean" since it directly reacts to state changes (when messages are added) and not to network events :)

For the other approach I had to add an additional `addAfterProcessingHandler` which is no longer needed with the store watcher :)